heron/uploaders/src/java/com/twitter/heron/uploader/hdfs/HdfsUploader.java

```diff
@@ -65,33 +66,32 @@ protected boolean isLocalFileExists(String file) {
   }
 
   @Override
-  public URI uploadPackage() {
+  public URI uploadPackage() throws UploaderException {
     // first, check if the topology package exists
     if (!isLocalFileExists(topologyPackageLocation)) {
-      LOG.info("Topology file " + topologyPackageLocation + " does not exist.");
-      return null;
+      throw new UploaderException(
+        String.format("Topology file %s does not exist", topologyPackageLocation));
     }
 
     // if the dest directory does not exist, create it.
     if (!controller.exists(destTopologyDirectoryURI)) {
       LOG.info("The destination directory does not exist; creating it.");
       if (!controller.mkdirs(destTopologyDirectoryURI)) {
-        LOG.severe("Failed to create directory: " + destTopologyDirectoryURI);
-        return null;
+        throw new UploaderException(
+            String.format("Failed to create directory: %s", destTopologyDirectoryURI));
       }
     } else {
       // if the destination file exists, write a log message
       LOG.info("Target topology file " + packageURI.toString() + " exists, overwriting...");
-
     }
 
     // copy the topology package to target working directory
     LOG.info("Uploading topology " + topologyPackageLocation
         + " package to target hdfs " + packageURI.toString());
 
     if (!controller.copyFromLocalFile(topologyPackageLocation, packageURI.toString())) {
-      LOG.severe("Failed to upload the package to:" + packageURI.toString());
-      return null;
+      throw new UploaderException(
+          String.format("Failed to upload the package to: %s", packageURI.toString()));
```

Comment:
agree. and this is important for https://github.com/twitter/heron/issues/1462#issuecomment-258550269